flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -274,6 +286,57 @@ class DataStreamOverAggregate(
       }
     result
   }
+  
+  def createTimeBoundedProcessingTimeOverWindow(inputDS: DataStream[Row]): DataStream[Row] = {
+
+    val overWindow: Group = logicWindow.groups.get(0)
+    val partitionKeys: Array[Int] = overWindow.keys.toArray
+    val namedAggregates: Seq[CalcitePair[AggregateCall, String]] = generateNamedAggregates
+
+    val index = overWindow.lowerBound.getOffset.asInstanceOf[RexInputRef].getIndex
+    val count = input.getRowType.getFieldCount
+    val lowerBoundIndex = index - count
+    
+    
+    val timeBoundary = logicWindow.constants.get(lowerBoundIndex).getValue2 match {
+      case bd: java.math.BigDecimal => bd.longValue()
+      case _ => throw new TableException("OVER Window boundaries must be numeric")
+    }
+
+     // get the output types
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+         
+    val result: DataStream[Row] =
+        // partitioned aggregation
+        if (partitionKeys.nonEmpty) {
+          
+          val processFunction = AggregateUtil.createTimeBoundedProcessingOverProcessFunction(
+            namedAggregates,
+            inputType,
+            timeBoundary)
+          
+          inputDS
+          .keyBy(partitionKeys: _*)
+          .process(processFunction)
+          .returns(rowTypeInfo)
+          .name(aggOpName)
+          .asInstanceOf[DataStream[Row]]
+        } else { // non-partitioned aggregation
+          val processFunction = AggregateUtil.createTimeBoundedProcessingOverProcessFunction(
```

Comment:
The `isPartitioned` parameter is not used in the `createTimeBoundedProcessingOverProcessFunction()`. So both calls are identical and we can move it outside of the condition and save a few LOCs.